### PR TITLE
Remove upper bound limit of astroid

### PR DESCRIPTION
**Pull request by auvipy**
The issue was fixed & released so removing the upper bound

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -248,9 +248,7 @@ def write_version(filename: str = str(AIRFLOW_SOURCES_ROOT / "airflow" / "git_ve
     "requests>=2.26.0",
 ]
 doc = [
-    # Astroid 2.12.* breaks documentation building
-    # We can remove the limit here after https://github.com/PyCQA/astroid/issues/1708 is solved
-    "astroid<2.12.0",
+    "astroid",
```
Comment (eladkal) [inline setup.py]:
```suggestion
    "astroid>=2.12.3",
```
Don't we need to bump to the version that contains the fix?
or at least skip the ones known to have the issue: `!=2.12.0, !=2.12.1, !=2.12.2` ?

Comment (auvipy) [reply setup.py]: done

**Review by potiuk — APPROVED** [review]

**Comment by auvipy** [comment]
thanks

**Comment by potiuk** [comment]
Thank you !

**Comment by vchiapaikeo** [comment]
Seeing this issue when trying to start breeze on latest w/ `breeze --python 3.9 --backend postgres start-airflow'`:

```
#0 6.661 ERROR: Cannot install apache-airflow[devel-ci]==2.6.0.dev0 because these package versions have conflicting dependencies.
#0 6.662
#0 6.662 The conflict is caused by:
#0 6.662     apache-airflow[devel-ci] 2.6.0.dev0 depends on astroid>=2.12.3
#0 6.662     The user requested (constraint) astroid==2.11.7
#0 6.662
#0 6.662 To fix this you could try to:
#0 6.662 1. loosen the range of package versions you've specified
#0 6.662 2. remove package versions to allow pip attempt to solve the dependency conflict
#0 6.662
#0 6.662 ERROR: ResolutionImpossible: for help visit https://pip.pypa.io/en/latest/topics/dependency-resolution/#dealing-with-dependency-conflicts
------
ERROR: failed to solve: executor failed running [/bin/bash -o pipefail -o errexit -o nounset -o nolog -c bash /scripts/docker/install_airflow.sh]: exit code: 1
Your image build failed. It could be caused by conflicting dependencies.
Run `breeze ci-image build --upgrade-to-newer-dependencies` to upgrade them.

Creating network "docker-compose_default" with the default driver
Pulling airflow (ghcr.io/apache/airflow/main/ci/python3.9:latest)...
latest: Pulling from apache/airflow/main/ci/python3.9
```

Do constraints for the devel-ci extra need to be updated somehow?

**Comment by potiuk** [comment]
Yes. They should update itself automatically (but it can a little time depending on the success of main build. 

As a temp solution you should follow the advice given (if you look closely:

> Your image build failed. It could be caused by conflicting dependencies.
> Run `breeze ci-image build --upgrade-to-newer-dependencies` to upgrade them.

**Comment by potiuk** [comment]
Yep. The constraints with updated astroid have just been pushed after main success: https://github.com/apache/airflow/commit/69b9d24d93a8fca00cd39e2b16b2c97ed26b0e86

So it should work now without need to add the flag

**Pull request merged**